packages/bikram-sambat/src/index.ts

```diff
@@ -363,4 +366,47 @@ export default class BikramSambat implements BikramSambatProps {
 
     return dayjs(this.adDate).isAfter(date, unit);
   }
+  /**
+   * Determines if a given BikramSambat or Date object falls between two specified dates, considering boundary inclusion.
+   *
+   * Supported boundaryInclusion values:
+   * - '()': Exclude both startDate and endDate
+   * - '[]': Include both startDate and endDate
+   * - '(]': Exclude startDate, include endDate
+   * - '[)': Include startDate, exclude endDate
+   *
+   * @param {BikramSambat | Date} startDate - The start date to compare against.
+   * @param {BikramSambat | Date} endDate - The end date to compare against.
+   * @param {ManipulateType} [unit='day'] - The unit of time for comparison. Defaults to 'day'.
+   * @param {BoundaryInclusionType} [boundaryInclusion='()'] - The boundary inclusion type for comparison. Defaults to '()'.
+   * @returns {boolean} True if the date is between startDate and endDate according to the specified boundaryInclusion, false otherwise.
+   */
+
+  isBetween(
+    startDate: BikramSambat | Date,
+    endDate: BikramSambat | Date,
+    unit: ManipulateType = 'day',
+    boundaryInclusion: BoundaryInclusionType = '()'
```

Comment:
add robust validation for boundary inclusion token using pattern matching and also add test for it